native/src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -86,5 +86,5 @@ export function * fetchCategory (dataContainer: DataContainer, action: FetchCate
 }
 
 export default function * (dataContainer: DataContainer): Saga<void> {
-  yield takeLatest('FETCH_CATEGORY', fetchCategory, dataContainer)
+  yield takeEvery('FETCH_CATEGORY', fetchCategory, dataContainer)
```

Comment:
This is necessary for navigating while refreshing. I don't think this has any negative side effects, but am not 100% certain. The alternative would be to add a new action just for refreshing.

Reply:
There isn't any other possibility to trigger two `FETCH_CATEGORY`s except navigating after refreshing, right?
What happens if a refresh is in progress, user is navigating to another category, the refresh finishes before the push finishes? Isn't there a route with outdated content then? Since this is probably a really rare edge case and not really harmful, I guess this is fine.

Reply:
Probably not an issue. Maybe it is possible to call FETCH_CATEGORY twice very fast. But this shouldn't do harm.

Reply:
This could be an issue if there are two resource cache updates. Then the app shows an error.
So probably 2 actions are safer.

Reply:
Setting the last update of the data container to the current time before something is actually loaded from the api in `loadCityContent` should resolve all issues. If it fails, it has to be set to the previous timestamp though probably.

Reply:
I'll try @klinzo's solution.
@maxammann How would a separate action resolve the problem you mentioned? The new action would pretty much call the same function as FETCH_CATEGORY.

Reply:
I think no new action is needed. But there are two reducers atm which contain a lot of duplicate code. They should be merged.

Reply:
I merged the respective files.